tests/unit_tests/crud/test_quality_matrix.py

```diff
@@ -214,3 +227,78 @@ def test_missing_fields_ratio():
     assert response == 100
     response = missing_fields_ratio({"doc_count": 5}, 10)
     assert response == 50
+
+
+def compare_lists_of_dict(expected, actually) -> bool:
```

Comment:
As discussed I would go with numpy arrays or xarray to hold the individual counts. that will save us all the handcrafted transpose code and tests.

If going with numpy I would use a dataclass e.g. like this

```python
@dataclass
class Counts:
    grouping: str # the name of the metadatum that defines the set of rows.
    total: numpy.ndarray # (M,) shaped array having the total number of documents for each row
    empty: numpy.ndarray # (M, N) shaped array with the number of documents in given row where given column metadatum is empty/missing 
    rows: list[str] # length M, a grouping of the documents according to a specific metadatum
    columns: list[str] # length N, the investigated other metadatums that are checked for missing/empty.
```
Eventually you would just have to flip the definition or row and column in the above snippet, but I guess it gets the point :P

Reply:
This would also be more or less the data structure that I would expose via rest. The frontend then should have all the information it needs to calculate percentages etc.
It is also generic enough to serve different quality matrices by having a different `grouping` and a different set  of `columns`.
Obviously having the `grouping` as part of `columns` should always yield 100% -right? :D

Reply:
Very good points - I added it to https://github.com/openeduhub/metaqs-main/issues/43